libs/fmt/fmt.c

```diff
@@ -418,6 +422,41 @@ DEFINE_PRIM(_I32, ogg_tell, _OGG);
 DEFINE_PRIM(_BOOL, ogg_seek, _OGG _I32);
 DEFINE_PRIM(_I32, ogg_read, _OGG _BYTES _I32 _I32);
 
+/* ----------------------------------------------- SOUND : MP3 ------------------------------------------------ */
+
+typedef struct _fmt_mp3 fmt_mp3;
+struct _fmt_mp3 {
+	char *bytes;
+	int size;
+	mp3dec_t dec;
+	mp3dec_frame_info_t info;
+	mp3d_sample_t pcm[MINIMP3_MAX_SAMPLES_PER_FRAME];
+};
+
+HL_PRIM fmt_mp3 *HL_NAME(mp3_open)( char *bytes, int size ) {
+	fmt_mp3 *o = (fmt_mp3*)hl_gc_alloc_raw(sizeof(fmt_mp3));

```

Comment:
You should not use allow_raw here, as this will make the GC scan the whole data for pointers, eventually creating false positive. What you want here is hl_gc_alloc_noptr, or alloc_finalizer if some memory needs to be freed when the decoder is no longer in-use.

Reply:
I'm not sure why, but if I use `alloc_noptr` or `alloc_finalizer` it throws `access_violation` when attempting to decode a frame.